imap/dav_db.c

```diff
@@ -256,14 +256,39 @@ struct sqldb_upgrade davdb_upgrade[] = {
   /* Don't upgrade to version 11.  We only jump to 11 on CREATE */
   /* Don't upgrade to version 12.  This was an intermediate Sieve DB version */
   /* Don't upgrade to version 13.  This was an intermediate Sieve DB version */
-  { 14, CMD_DBUPGRADEv14, &sievedb_upgrade },
+  { 14, CMD_DBUPGRADEv14,
```

Comment:
Oh man - I hate this complexity level :(  We should just require sieve to be included.

Reply:
I think my last commit makes this "better" for some value of "better".  However, it does guarantee that the Sieve table schema is correct if some installation was at v12/v13 and disabled Sieve for v14 (highly unlikely, but possible).